cmd/conduit-pipeline-check/main.go

```diff
@@ -0,0 +1,123 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"os/signal"
+	"time"
+
+	apiv1 "github.com/conduitio/conduit/proto/gen/api/v1"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+const (
+	exitCodeOk = iota
+	exitCodeErr
+	exitCodeInterrupt
+)
+
+func main() {
+	ctx := cancelOnInterrupt(context.Background())
+
+	exitcode, err := checkPipeline(ctx)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+	}
+	os.Exit(exitcode)
```

Comment:
Here `exitcode` will always be `exitCodeErr` if an error is returned or `exitCodeOk` if it's not, right? We could simplify this:
```suggestion
	err := checkPipeline(ctx)
	if err != nil {
		fmt.Fprintln(os.Stderr, err)
		os.Exit(exitCodeErr)
	}
```
The nice thing about this is we don't call `os.Exit(0)` which just terminates the program right away without calling any deferred functions. Not that it matters here, but you know, good practice 🙃

Reply:
yup this makes sense. will make the change, originally i started with just an error as well.

Reply:
fixed in bc01d7c425731ccb18893dd3447adcec34012f64